### PR TITLE
json method for jsonp

### PR DESCRIPTION
**Pull request by sebjwallace**
```javascript
Ajax().json('http://jsonplaceholder.typicode.com/posts/1', function(data){
    // callback
})
```

When requesting json while avoiding cross-origin issues jsonp is the only choice. It only needs the url and callback.

I was unable to implement a test using the localhost api so I used jsonplaceholder
```
http://jsonplaceholder.typicode.com/posts/1
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -222,6 +222,16 @@ ajax().post('/api/users', { slug: 'john' }).always(function(response, xhr) {
 })
 ```
 
+### `json(url, function(){/* callback */})`
+
+> Get data as a JSON object as jsonp.
+
+```js
+ajax().json('/api/users', function(users){
```
Comment (fdaciuk) [inline README.md]: We could try using `.then` method instead callback =)

File: test/unit/ajax-json.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+;(function (should, expect, Ajax, ajax) {
+  'use strict'
+
+  describe('#AJAX - Test `json` method', function () {
+    var request
+    beforeEach(function () {
+      request = new Ajax()
```
Comment (fdaciuk) [inline test/unit/ajax-json.test.js]: -1
`new Ajax()` is deprecated. Use just `ajax()`, please.

File: test/unit/ajax-json.test.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    })
+    it('Should return data about `posts`', function (done) {
+      request.get('http://jsonplaceholder.typicode.com/posts/1')
```
Comment (fdaciuk) [inline test/unit/ajax-json.test.js]: Could you write tests for `json` method, instead `get`?

File: test/unit/ajax-json.test.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        .done(function (response) {
```
Comment (fdaciuk) [inline test/unit/ajax-json.test.js]: -1
`.done` is deprecated =)

File: src/ajax.js [hunk 5]
```diff
@@ -180,6 +180,22 @@
       )
     }
 
+    $public.json = function json (url, callback) {
```
Comment (fdaciuk) [inline src/ajax.js]: I think `jsonp` is clearer. =)

**Comment by fdaciuk** [comment]
Hey @sebjwallace! Thanks for your contribution :smiley: 
But pull request should be sent against `dev` branch. 
If you have any doubt, read [`CONTRIBUTING.md`](https://github.com/fdaciuk/ajax/blob/dev/CONTRIBUTING.md) file, please =)

Thanks again! :+1:

**Comment by sebjwallace** [comment]
Thanks for the feedback. Looking back, I'm surprised how far off-base I was writing the test/s. I'll send the next commit to the dev brach.

**Comment by fdaciuk** [comment]
Thanks :dancer: :sparkles:

**Pull request closed without merging**